[bugfix] QA and fixes #19

trombik opened the pull request:
no functionality changes

Commit:
QA

Commit:
bump centos version

Commit:
fix #19

Commit:
Merge branch 'master' into QA

Commit:
QA

Commit:
use ansible_distribution_major_version

instead of ansible_distribution_version.

minor releases are supposed not to affect the role variables

Commit:
workaround Debian bug 863888

also, remove hard dependecny (this is supposed to be soft)

Commit:
update README

Commit:
no dependency in `meta/main.yml`

mitsururike commented on README.md:
```diff
@@ -46,15 +47,15 @@ None
+| `__java_packages` | `["java-1.8.0-openjdk", "java-1.8.0-openjdk-devel"]` |
 
-## Ubuntu-14.04
+## Ubuntu-14
```
does the role support Ubuntu-14.10 as well?

trombik replied:
it is supposed to do so.

mitsururike replied:
meta/main.yml should be updated.

trombik replied:
i have not confirmed, nor we have a box for the version

trombik replied:
also, i do not see a point to support versions whose support term is shorter

mitsururike replied:
it looks like it supports Ubuntu-14.10 as the title of the section is "Ubuntu-14".
if it does not support 14.10 officially, the title should be "Ubuntu-14.04".

trombik replied:
`meta/main.yml` clearly states supported (and tested) versions. in general, i have been trying to make roles version-agnostic. why are you trying to be so specific? according to your suggestion, we must maintain every version of Debian, such as 8.6, 8.7, and so on just for nothing. i do not see the point.

mitsururike replied:
Debian 8.6, 8.7 correspond to Ubuntu 14.04.1, 14.04.2.
`Debian-8` or `Ubuntu-14.04` implies that the OS is up to date.
`Ubuntu-14` means `Ubuntu-14.04 and Ubuntu-14.10` which are the different releases.
`Ubuntu-14` may be misleading the users that the role is tested or does suuport Ubuntu-14.10.

BTW, Ubuntu 14.10 has an official package of openjdk8 and ppa https://launchpad.net/~openjdk-r/+archive/ubuntu/ppa is available for Ubuntu 14.04.

trombik replied:
would you kindly answer the questions? ignoring the points i made and keep pushing your opinions is not a discussion.

you are suggesting rather lots of changes in all the repositories.

mitsururike replied:
i think i have answered the questions and explained my points.
i do not understand which points i have ignored.
and also i do not understand what `you are suggesting rather lots of changes in all the repositories.` means.

trombik replied:
why are you trying to change the meaning of the sections? you are not happy with `meta/main.yml` even when it says which version is supported. you are saying mere section names extracted from file names where the listed variables come from must match supported versions. the sections are there for reader's convenience. when i did research on `ansible` and how people write roles, lots of them do not explain default values of variables. they simply say "see files for defaults", which is just annoying. I do not think I have promised the sections names in question are supported versions.

why are you trying to be so specific while I am trying to write generic roles that could work for non-supported versions? what is the reason? according to your suggestions, all the `README.md` should show specific supported versions. if "Ubuntu-14 means Ubuntu-14.04 and Ubuntu-14.10 which are the different releases", writing "RedHat" in `README.md` is wrong. neither "RedHat 7". it must have been "RedHat 7.3".

trombik replied:
> i do not understand what you are suggesting rather lots of changes in all the repositories. means.

all in all, to enforce your suggestion, we need to re-write the rules. all of `README.md` must be updated. for what? for the completeness? as I wrote, I see no points.

mitsururike replied:
| versions in README.md | versions in meta/main.yml | OK/NG | explanation |
|--------------------------------|----------------------------------|-----------|---------------|
| N/A | some versions | OK | because it is obvious that versions described in meta/main.yml are supported versions |
| some versions described in meta/main.yml | some versions | OK | because they matched and no conflict |
| some versions | N/A | NG | because meta/main.yml must have supported versions |
| some versions NOT matched with those in meta/main.yml | some versions | NG | because users could not know which versions are supported |

i do not think it is a good idea to send users a message that the role might work for non-supported versions via README.md as we do not test it at all.

trombik replied:
i do not think we are communicating because i asked why you are trying to change the meaning of the sections only to be ignored. in a discussion or a conversation, "why" question is always followed by "because". the second question seems to get an answer, but is not. your answer does not explain why your are nitpicking.

despite the unclear answers, your strong opposition to portable roles has been understood. here is logical consequences from your opinion; why don't you oppose an idea to support platforms, as someone expressed in the past, that we do not use? that is just a waste of time. it is much easier to write non-portable code. we should bails out if the platform is not supported, or `uname -r` is not `Linux`, because we have not tested and running a role on unsupported platforms would kill kittens.

in short, you are suggesting new policy to be implemented, not pointing out defects, bugs, or anything in the role itself. I really appreciate if you propose elsewhere, not in this PR, because the PR is meant to solve issues in the development. if you want new policy, this is wrong place.

if you are okay with other part of the code, please approve. feel free to create an issue with your proposals. the PR is blocking other PR and the development. thank you for being productive.

mitsururike replied:
reading your questions, i thought you did not understand my points because my explanation was not clear and i tried to explain them.
and i did not think i had to answer with "because".
here are my answers.
> why are you trying to change the meaning of the sections?

because i am not. i pointed out that "Ubuntu-14" can be understood as "Ubuntu-14.04 and Ubuntu-14.10" by users. i know you do not intend to send a wrong message, but it could be.


> why are you trying to be so specific while I am trying to write generic roles that could work for non-supported versions? what is the reason?

because i think the role should be generic but documents should not describe the untested versions.
the table above explains the detail.

i thought it is obvious and not necessary to make it a rule once you understand my points.

trombik replied:
> i thought it is obvious and not necessary to make it a rule once you understand my points.

clearly, as shown in the discussion, it is not obvious.

mitsururike commented on README.md:
```diff
@@ -46,15 +47,15 @@ None
 | `__java_packages` | `["oracle-java8-installer"]` |
 
-## Ubuntu-16.04
+## Ubuntu-16
```
does the role support Ubuntu-16.10 as well?

trombik replied:
it is supposed to do so.

mitsururike replied:
meta/main.yml should be updated.

mitsururike reviewed (approved):
approved not to block the other PRs.
section names of REDAME.md is still in discussion.
a rule should be created in case of no agreement.

Commit:
make everyone happy

as requested at:
https://github.com/reallyenglish/ansible-role-java/pull/23#discussion_r132084161
